report/src/main/webapp/assets/js/newComparison/newComparison.js

```diff
@@ -0,0 +1,698 @@
+
+			if (sizes) {
+				var heightA = sizes.aTo - sizes.aFrom;
+				var heightB = sizes.bTo - sizes.bFrom;
+
+				newDifferenceA.width = newDifferenceB.width = newDifferenceContextA.width = newDifferenceContextB.width = imgSizeA.width;
+				newDifferenceA.height = newDifferenceContextA.height = heightA;
+				newDifferenceB.height = newDifferenceContextB.height = heightB;
+
+				newDifferenceB.style.left = '52%';
+
+				difs.appendChild(newDifferenceA);
+				difs.appendChild(newDifferenceB);
+
+				newDifferenceA.style.top = sizes.aFrom * proc + '%';
+				newDifferenceB.style.top = sizes.bFrom * proc + '%';
+
+				newDifferenceA.style.height = (sizes.aTo - sizes.aFrom) * proc + '%';
+				newDifferenceB.style.height = (sizes.bTo - sizes.bFrom) * proc + '%';
+
+				var dataA = contextA.getImageData(0, sizes.aFrom, imgSizeA.width, heightA);
+				var dataB = contextB.getImageData(0, sizes.bFrom, imgSizeA.width, Math.max(heightB, 1));
+				var imageData;
+
+				if (dataA.height > dataB.height) {
+					imageData = convertDifferenceData(dataA, dataB);
+				} else {
+					imageData = convertDifferenceData(dataB, dataA);
+				}
+
+				newDifferenceContextA.putImageData(imageData, 0, 0, 0, 0, imgSizeA.width, heightA);
+				if (heightB > 0) {
+					newDifferenceContextB.putImageData(imageData, 0, 0, 0, 0, imgSizeA.width, heightB);
+				}
+
+				difsA.push(newDifferenceA);
+				difsB.push(newDifferenceB);
+			}
+		}
+	}
+
+	function convertDifferenceData(data1, data2) {
+		for (var index = 0; index < data1.data.length; index += 4) {
+			if (typeof data2.data[index] !== 'undefined') {
+				if (data1.data[index] !== data2.data[index] || data1.data[index + 1] !== data2.data[index + 1] || data1.data[index + 2] !== data2.data[index + 2]) {
+					data1.data[index] = 255;
+					data1.data[index + 1] = 0;
+					data1.data[index + 2] = 0;
+					data1.data[index + 3] = 127;
+				} else {
+					data1.data[index] = 0;
+					data1.data[index + 1] = 0;
+					data1.data[index + 2] = 0;
+					data1.data[index + 3] = 0;
+				}
+			} else {
+				data1.data[index] = 255;
+				data1.data[index + 1] = 0;
+				data1.data[index + 2] = 0;
+				data1.data[index + 3] = 127;
+			}
+		}
+
+		return data1;
+	}
+
+	function loadImage(url) {
+    debug && console.log('load image: ', url);
+		return new Promise(function(resolve, reject) {
+			var image = new Image();
+
+			image.onload = function() {
+        debug && console.log('onload');
+				resolve(image);
+			}
+			image.onerror = function() {
+        debug && console.log('onerror');
+				reject();
+			}
+
+			image.src = url;
+		});
+	}
+
+	function rgb2hex(r, g, b) {
+		return ('0' + r.toString(16)).slice(-2) + ('0' + g.toString(16)).slice(-2) + ('0' + b.toString(16)).slice(-2);
+	}
+
+  function simplifyImage(imageData) {
+    var imageWidth = imageData.width;
+    var offset = 0;
+    var data = imageData.data;
+    var simplifiedList = [];
+    var previousLine;
+
+    for (var y = 0; y < data.length / imageWidth / 4; y ++) {
```

Comment:
I think `data.length / (imageWidth * 4)` would look clearer

Also - why 4? Magic variable? ✨
I can see it being used throughout the function. Consider putting it in var with descriptive name.

Reply:
It's because of way data from image is provided. It reads all pixels and merges it into single array:
[r0,g0,b0,a0,r1,g1,b1,a1,...] r0 = red value from first pixel, r1 = red value from second pixel etc.
It can be named tho. I'll do it.